### PR TITLE
feat(query): trailing comma functionality

### PR DESCRIPTION
**Pull request by Arhanace**

### CODE REVIEW

File: prometheus/src/test/scala/filodb/prometheus/parse/ParserSpec.scala [hunk 1]
```diff
@@ -465,6 +465,11 @@ class ParserSpec extends AnyFunSpec with Matchers {
     parseSuccessfully("timestamp(some_metric)")
     parseError("timestamp(some_metric[5m])") // reason : Expected instant vector, got range vector
     parseError("timestamp(some_metric, hello)") // reason : Expected only 1 arg, got 2
+
+    // Trailing Commas
+    parseSuccessfully("sum without(and, by, avg, count, alert, annotations,)(some_metric)")
+    parseSuccessfully("sum without(and, by, avg, count, alert, annotations, )(some_metric)")
+    parseSuccessfully("sum by(and, by, avg, count, alert, annotations, )(some_metric)")
```
Comment (alextheimer) [inline prometheus/src/test/scala/filodb/prometheus/parse/ParserSpec.scala:472]: Could you add a couple for binary joins (i.e. `+`/`-`/`and`, etc), as well? The analogous clauses are `on` and `ignoring`, and the label lists used for each are called "`labelNameList`" in the grammar (I am not sure offhand why we allow more complex expressions in an aggregation's clause).

**Review by alextheimer — CHANGES_REQUESTED** [review]

File: prometheus/src/test/scala/filodb/prometheus/parse/ParserSpec.scala (hunk 1 above)
Comment (alextheimer) [reply prometheus/src/test/scala/filodb/prometheus/parse/ParserSpec.scala:472]: Ah I see I was confused -- `labelNameList` is the one you already updated, and `parameterList` has nothing to do with a `by`/`without` clause:
```
aggregation
    : AGGREGATION_OP parameterList
    | AGGREGATION_OP (by | without) parameterList
    | AGGREGATION_OP parameterList ( by | without)
    ;
by:      BY labelNameList;
without: WITHOUT labelNameList;
```

Cool, I thought we needed more updates to the grammar but that's not the case.

File: prometheus/src/test/scala/filodb/prometheus/parse/ParserSpec.scala [hunk 2]
```diff
@@ -465,6 +465,21 @@ class ParserSpec extends AnyFunSpec with Matchers {
...
+
+    // Trailing Commas in Binary Joins
+    parseSuccessfully("foo and ignoring(test,blub,) bar")
+    parseSuccessfully("foo and ignoring(test,blub, ) bar")
+
+    parseSuccessfully("foo / on(test,blub, ) group_left(bar) bar")
+    parseSuccessfully("foo / ignoring(test,blub,) group_left(blub) bar")
+
+    parseSuccessfully("foo - on(test,blub,) group_right(bar,foo,) bar")
```
Comment (alextheimer) [inline prometheus/src/test/scala/filodb/prometheus/parse/ParserSpec.scala:481]: Excellent call with these `group` clauses 👍

**Review by alextheimer — APPROVED** [review]

File: prometheus/src/test/scala/filodb/prometheus/parse/ParserSpec.scala [hunk 3]
```diff
@@ -465,6 +465,21 @@ class ParserSpec extends AnyFunSpec with Matchers {
...
     parseError("timestamp(some_metric[5m])") // reason : Expected instant vector, got range vector
     parseError("timestamp(some_metric, hello)") // reason : Expected only 1 arg, got 2
+
+    // Trailing Commas
+    parseSuccessfully("sum without(and, by, avg, count, alert, annotations,)(some_metric)")
+    parseSuccessfully("sum without(and, by, avg, count, alert, annotations, )(some_metric)")
+    parseSuccessfully("sum by(and, by, avg, count, alert, annotations, )(some_metric)")
+
+    // Trailing Commas in Binary Joins
+    parseSuccessfully("foo and ignoring(test,blub,) bar")
```
Comment (yu-shipit) [inline prometheus/src/test/scala/filodb/prometheus/parse/ParserSpec.scala:475]: should we also add tests for the no trailing comma case?

**Comment by amolnayak311** [comment]
Please follow https://github.com/filodb/FiloDB/blob/develop/CONTRIBUTING.md for PR title and summarize the PR changes. Also congratulation on your first PR.

**Pull request merged**